get-started/code-experiments/src/prepare.py

```diff
@@ -0,0 +1,108 @@
+import struct
+import numpy as np
+import gzip
+import os
+from util import load_params
+
+
+def mnist_images_idx_to_array(images_filename):
+    images_f = gzip.open(images_filename, mode="rb")
+    images_f.seek(0)
+    magic = struct.unpack('>I', images_f.read(4))[0]
+    if magic != 0x00000803:
+        raise Exception(f"Format error: Need an IDX3 file: {images_filename}")
+    n_images = struct.unpack('>I', images_f.read(4))[0]
+    n_row = struct.unpack('>I', images_f.read(4))[0]
+    n_col = struct.unpack('>I', images_f.read(4))[0]
+
+    n_bytes = n_images * n_row * n_col  # each pixel is 1 byte
+
+    images_data = struct.unpack(
+        '>' + str(n_bytes) + 'B', images_f.read(n_bytes))
+
+    images_array = np.asarray(images_data, dtype='uint8')
+    images_array.shape = (n_images, n_row, n_col)
+
+    return images_array
+
+
+def mnist_labels_idx_to_array(labels_filename):
```

Comment:
same here - code should be simpler

Reply:
if we want to have some meaningful prepare step - I would apply some augmentations or some other image transformations